Softeq.XToolkit.Common.Droid/Extensions/StringExtensions.cs

```diff
@@ -12,48 +14,86 @@ namespace Softeq.XToolkit.Common.Droid.Extensions
     public static class StringExtensions
     {
         /// <summary>
-        ///     Allows to apply multiple spans to the whole string.
+        ///     Allows to apply multiple spans to part of the string.
         /// </summary>
         /// <returns>The spannable string with all spans apllied.</returns>
         /// <param name="unformattedString">Unformatted string.</param>
+        /// <param name="textRange">Instance containing the starting index and the length of the range to which spans shall be applied.</param>
         /// <param name="spans">An array of spans.</param>
-        public static SpannableString FormatSpannable(this string unformattedString, params Object[] spans)
+        /// <exception cref="T:System.ArgumentNullException">
+        ///     <paramref name="unformattedString"/> and <paramref name="textRange"/> parameters cannot be <see langword="null"/>.
+        /// </exception>
+        /// <exception cref="T:System.ArgumentException">
+        ///     <paramref name="spans"/> list cannot be empty.
+        /// </exception>
+        /// <exception cref="T:System.ArgumentOutOfRangeException">
+        ///     Starting index of the <paramref name="textRange"/> must be less than the size of the string.
+        ///     - and -
+        ///     Length of the <paramref name="length"/> must be less than the size of the string.
+        ///     - and -
+        ///     End index of the <paramref name="textRange"/> must be less than the size of the string.
+        /// </exception>
+        public static SpannableString FormatSpannable(
+            this string unformattedString,
+            TextRange textRange,
+            params Object[] spans)
         {
+            if (unformattedString == null)
+            {
+                throw new ArgumentNullException($"{nameof(unformattedString)} cannot be null");
```

Comment:
can be simplified as `new ArgumentNullException(nameof(unformattedString))` as a result:

```
System.ArgumentNullException: Value cannot be null.
Parameter name: unformattedString
...
```

Reply:
Reworked